resources/assets/js/src/components/chart/Chart.tsx

```diff
@@ -212,28 +221,23 @@ export default class Chart extends React.Component<IProps, IState> {
     }
 
     private loadData(dateFrom: string, dateTo: string, direction: string = null): void {
+        this.setError();
         const url = URLUtil.generateURLByDates(this.props.url, dateFrom, dateTo);
         let dataMeta;
 
         axios.get(url).then((response: any) => {
             const newData = this.processResponse(response);
-            if (!newData.length || newData[0].date === this.state.initialDate) {
+            if (!this.checkData(newData)) {
                 return;
             }
 
             if (direction) {
                 let data = ArrayUtil.destructureDataArrays(direction, this.state.data, newData);
                 data = ArrayUtil.removeDuplicities(data);
-                dataMeta = {
-                    firstDate: data[0].date,
-                    lastDate: data[data.length - 1].date,
-                };
+                dataMeta = { firstDate: data[0].date, lastDate: data[data.length - 1].date };
                 this.setState({ data, dataMeta });
             } else {
-                dataMeta = {
-                    firstDate: newData[0].date,
-                    lastDate: newData[newData.length - 1].date,
-                };
+                dataMeta = { firstDate: newData[0].date, lastDate: newData[newData.length - 1].date };
```

Comment:
what if newData[newData.length - 1] is undefined?

Reply:
that is handled by checkData function up to this code